src/shm.h

```diff
@@ -1,7 +1,11 @@
+#include <stdbool.h>
+
 struct shm {
-	char *buf;
-	char  name[PATH_MAX];
-	char  storage[32];
+	char  *buf;
+	char   name[PATH_MAX];
+	char   storage[32];
+	size_t buf_size;
```

Comment:
This element of the struct ends up duplicating `priv.sz` from `src/unix/shm.c:15`, which is stored in `shm.storage`, and I think a good clean-up would be to use this over `p.sz`